### ISSUE TITLE
Multiple servicemonitors not processed if one is faulty

### ISSUE THREAD

**Issue by R0peE** [prometheus-operator/prometheus-operator#4201]
**What happened?**

Additional servicemonitors are not procssed by the operator in case one of them contains reference to non existent resource (eg. configmap, secret etc.)

**Did you expect to see something different?**

Skip the faulty servicemonitor and continue processing the rest

**How to reproduce it (as minimally and precisely as possible)**:

Create a servicemonitor with http basic auth parameters pointing to a non-existent secret, and add additional servicemonitors

**Environment**

* Prometheus Operator version:

v.0.35.1-2

* Kubernetes version information:

OpenShift Version
4.6.25

* Prometheus Operator Logs:

```
E0805 13:35:00.700059       1 operator.go:984] Sync "mcs-monitoring/k8s" failed: creating config failed: could not generate basicAuth for servicemonitor (...) unable to fetch username secret "(...)": secrets "(...)" not found

```

**Comment by fpetkovski** [comment]
Are `Prometheus` resources which do not select the faulty ServiceMonitors affected as well?

**Comment by R0peE** [comment]
Hi Filip,

Thx for the reply, it seems to also affect other resources. While adding a "faulty" ServiceMonitor i have also added a PrometheusRule, and it is also not being processed.

**Comment by fpetkovski** [comment]
Ok, thanks for the information. As you are using a fairly old version of the operator, my first suggestion would be to try the latest version since this issue might already been fixed.

**Comment by simonpasquier** [comment]
Closing since this got fixed in #3445 (available in v0.42 and later).
